crates/telemetry/src/logs.rs

```diff
@@ -30,20 +29,16 @@ fn app_log_to_otel(buf: &[u8]) {
         return;
     }
 
-    let logger = global::logger_provider().logger("spin");
+    let logger = LOGGER.get().unwrap();
```

Comment:
Is panicking here the right way to go? Or would it make sense to return early if the logger isn't set (and perhaps log a warning)?

Reply:
If we `.expect()` when we set the logger than I think we should unwrap here.

Otherwise if we allow ourselves to fail to set a logger than I think we should do what you said here and return early. The annoying thing then is we should probably do the thing where you log at warn once about this failing and then at trace the rest of the time b/c this is going to be very noisy.

Reply:
I agree this probably shouldn't panic. I am ambivalent as to whether it needs to log instead given that this will log to tracing perfectly happily even if this fails.